### PR TITLE
Climbing exclusion update - Prettier pom.xml

### PR DESCRIPTION
**Pull request by Jakllp**
Listener:
- Better ignoreJump-System
    - noJump() -> looks for part of string in Material.name() for different types of e.g. fence
    - Used isPassable() to exclude everything that a player can walk through (this includes liquids)
- You can now get off of "softblocks" (Blocks with a full Hitbox even though entities are slightly lower when standing on them)
    - Previously: Just checking current block hitbox-height vs frontblock hitbox-height -> couldn't get off of e.g. soul_sand
    - Now: Also checking current height of vehicle -> if it is below the hitbox of the frontblock -> climb
    - Also: Had to make a list of softblocks - if a vehicle is in a softblock and driving onto another softblock it would climb, resulting in "hovering" (similar to steps before the last climbing update) - this prevents that
- Fixed getting driving passenger (could cause null-pointer)
- Removed some Legacy-Materials

pom.xml:
- Basically just a readability update - fixed indentation

plugin.yml:
- Version number is now controlled by pom.xml

Bump to 1.16-R1.1.2

### CODE REVIEW

File: src/main/java/com/useful/ucars/uCarsListener.java [hunk 1]
```diff
@@ -2069,46 +2063,24 @@ public void run() {
 		};
 	}*/
 	
+	public boolean noJump(String bName) {
+		for(String str:ignoreJump) {
+			if(bName.contains(str)) {return true;}
```
Comment (storm345) [inline src/main/java/com/useful/ucars/uCarsListener.java:2068]: Is this quite foolproof? Is there some edge case where maybe you don't want to jump for a "STEP" or something but do for a "DOUBLE_STEP"? (Bad example since it doesn't apply, but a similar idea where on material name is included in another)

Comment (storm345) [reply src/main/java/com/useful/ucars/uCarsListener.java:2068]: It may be okay, but changing the 'it matches' check to an 'it contains' checks seems problematic

Comment (storm345) [reply src/main/java/com/useful/ucars/uCarsListener.java:2068]: Maybe check GRASS and GRASS_BLOCK?

File: src/main/java/com/useful/ucars/uCarsListener.java [hunk 2]
```diff
@@ -2069,46 +2063,24 @@ public void run() {
...
-		ignoreJump.add(Material.ACACIA_FENCE_GATE.name());
-		ignoreJump.add(Material.BIRCH_FENCE.name());
-		ignoreJump.add(Material.BIRCH_FENCE_GATE.name());
-		ignoreJump.add(Material.JUNGLE_FENCE.name());
-		ignoreJump.add(Material.JUNGLE_FENCE_GATE.name());
-		/*ignoreJump.add("CARPET"); // carpet
-*/		
+		ignoreJump.add("WALL");
+		ignoreJump.add("FENCE");
+		ignoreJump.add("GATE");
```
Comment (storm345) [inline src/main/java/com/useful/ucars/uCarsListener.java:2077]: Will the car now jump for grass plants?

File: src/main/java/com/useful/ucars/ucars.java [hunk 3]
```diff
@@ -170,6 +171,11 @@ public void run() {
 	@Override
 	public void onEnable() {
 		plugin = this;
+		
+		//bStats
+		int pluginId = 11901; // <-- Replace with the id of your plugin!
+        Metrics metrics = new Metrics(this, pluginId);
```
Comment (storm345) [inline src/main/java/com/useful/ucars/ucars.java]: Please can we keep bStats out of the master uCars repo, I'm not a huge fan of non-consensual analytics data being sent to a third party - especially through a plugin ID managed by somebody else

**Review by storm345 — CHANGES_REQUESTED** [review]
Mostly looks good! If you can remove the bStats part, and check that the changing of ignoreJump to contains doesn't have any problematic edge cases (Eg. ignore jump for a 'GRASS' but not for a 'GRASS_BLOCK' which contains 'GRASS' or similar) then I will approve :)

**Comment by Jakllp** [comment]
About the jumping:
The noJump() function is the only function using the ignoreJump array.
It used to handle basically everything but now it's just handling 1 1/2 blocks (so walls, fences, gates) that have multiple materials (oak, birch etc) which is why the contains check works.

For grass etc I added an "isPassable()"-check to the checks performed before a jump. This excludes everything that a normal player can walk through (Grass, Flowers etc) - [here](https://github.com/Jakllp/ucars/blob/master/src/main/java/com/useful/ucars/uCarsListener.java#L1192) and [here](https://github.com/Jakllp/ucars/blob/master/src/main/java/com/useful/ucars/uCarsListener.java#L1197) - the latter is being used to jump up blocks that have grass standing on them

About bStats:
I'll go ahead and remove it again - I get where you are coming from, I'm not too much of a fan of "opt-out" systems myself but I thought it might be cool to see how many servers are actually using the plugin.
I respect your decision though.

File: src/main/java/com/useful/ucars/uCarsListener.java (hunk 1 above)
Comment (Jakllp) [reply src/main/java/com/useful/ucars/uCarsListener.java:2068]: See main comment

File: src/main/java/com/useful/ucars/uCarsListener.java (hunk 2 above)
Comment (Jakllp) [reply src/main/java/com/useful/ucars/uCarsListener.java:2077]: See main comment

**Comment by storm345** [comment]
> About the jumping:
> The noJump() function is the only function using the ignoreJump array.
> It used to handle basically everything but now it's just handling 1 1/2 blocks (so walls, fences, gates) that have multiple materials (oak, birch etc) which is why the contains check works.
> 
> For grass etc I added an "isPassable()"-check to the checks performed before a jump. This excludes everything that a normal player can walk through (Grass, Flowers etc) - [here](https://github.com/Jakllp/ucars/blob/master/src/main/java/com/useful/ucars/uCarsListener.java#L1192) and [here](https://github.com/Jakllp/ucars/blob/master/src/main/java/com/useful/ucars/uCarsListener.java#L1197) - the latter is being used to jump up blocks that have grass standing on them
> 
> About bStats:
> I'll go ahead and remove it again - I get where you are coming from, I'm not too much of a fan of "opt-out" systems myself but I thought it might be cool to see how many servers are actually using the plugin.
> I respect your decision though.

Yeah if you're sure that there isn't some edge cases missed with the no-jump then it's fine. I understand the change.

And thanks regarding bStats

**Comment by Jakllp** [comment]
All cases in the ignoreJump list are covered and even some more so it should be fine
